runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsOverlay.java

```diff
@@ -176,16 +193,65 @@ else if (plugin.getRoomBlacklist().contains(room.getName().toLowerCase()))
 					}
 
 					name = room == RaidRoom.UNKNOWN_PUZZLE ? "Unknown" : room.getName();
-
+					recommendedItems.add(room.getRecommendedItem());
 					panelComponent.getChildren().add(LineComponent.builder()
-						.left(room.getType().getName())
+						.left(showItems ? "" : room.getType().getName())
 						.right(name)
 						.rightColor(color)
 						.build());
 					break;
 			}
 		}
 
-		return panelComponent.render(graphics);
+		//if size is 1 the only thing in it is -1 which isnt an item
+		if (!showItems || recommendedItems.size() == 1)
+		{
+			return panelComponent.render(graphics);
+		}
+
+		Dimension mainPanelDimensions = panelComponent.render(graphics);
+		FontMetrics metrics = graphics.getFontMetrics();
+		int fontHeight = metrics.getHeight();
+		String fontName = metrics.getFont().getFontName();
+
+		// adjusts the size of the images in order to have them layout correctly with the different sized fonts
+		int imageSize = fontName.equals("RuneScape Small") ? SMALL_IMAGE_SIZE
+				: fontName.equals("RuneScape Standard") ? REGULAR_IMAGE_SIZE : BOLD_IMAGE_SIZE;
+
+		// adjust the location the images will be rendered on the parent component based on whats already there
+		int verticalOffset = (config.ccDisplay() ? fontHeight * 2 : fontHeight) + ComponentConstants.STANDARD_BORDER;
+
+		itemImages.getChildren().clear();
+		itemImages.setBorder(new Rectangle(0, 0, 0, 0));
+		itemImages.setPreferredLocation(new Point(ComponentConstants.STANDARD_BORDER, verticalOffset));
+		itemImages.setBackgroundColor(null);
+		itemImages.setWrapping(2);
+		itemImages.setOrientation(ComponentOrientation.HORIZONTAL);
+
+		for (int item : recommendedItems)
+		{
+			final BufferedImage image = item != -1 ? getImage(item, imageSize) : null;
```

Comment:
Following the comment above, this is simplified now that `-1` is not contained in the item set.
```suggestion
			final BufferedImage image = getImage(item, imageSize);
```